### PR TITLE
WEBSITE-245 [OGM] Improve contribute page

### PR DESCRIPTION
**Pull request by DavideD**
- Add link to JBoss CLA
- Add link to Eclipse and IntelliJ IDEA templates
- Add Java header for OGM
- Add Javadoc guide lines

https://hibernate.atlassian.net/browse/WEBSITE-245

### CODE REVIEW

File: ogm/contribute.adoc [hunk 1]
```diff
@@ -95,12 +100,9 @@ If you just want to build the documentation only,
 run it from the [filename]+hibernate-ogm-documentation/manual+ subdirectory.
 ====
 
-=== How to contribute code effectively
+=== How to contribute the code effectively
```
Comment (hferentschik) [inline ogm/contribute.adoc]: Is there not a 'to' missing "How to contribute to the code effectively". That said, why not just "How to contribute"

Comment (DavideD) [inline ogm/contribute.adoc]: "How to contribute" is the main title. The code is not the only way you can contribute.
"How to contribute code effectively" is correct, I'll remove the "the".

File: ogm/contribute.adoc [hunk 2]
```diff
@@ -97,10 +102,7 @@ run it from the [filename]+hibernate-ogm-documentation/manual+ subdirectory.
 
 === How to contribute code effectively
```
Comment (hferentschik) [inline ogm/contribute.adoc:87]: I still don't like 'effectively'. I find it very odd, but maybe that's just me.

**Comment by hferentschik** [comment]
I find the whole section "== How to get help" misplaced in this page. Why is it there at all? And then the "How to contribute" section starts with 'Welcome!', which I expect somewhere at the top of a page.

**Comment by DavideD** [comment]
> I find the whole section "== How to get help" misplaced in this page. Why is it there at all? And then the "How to contribute" section starts with 'Welcome!', which I expect somewhere at the top of a page.

Right, they should probably be merged in one section "How to contribute". But I suspect that a user will look for a section "How to get help".

**Comment by hferentschik** [comment]
> But I suspect that a user will look for a section "How to get help". 

Sure, you can have this information. My point is, that i find it odd where it is now and how it flows with the rest of the page. The page is called "How to contribute to OGM" and then the first section is "how to get help", followed by the actual "how to contribute". IMO you have three choices:
- Change the order and put it at the end possible renaming it to "how to get further help?"
- Add this to the FAQ as a question "How to get help"
- Create a dedicated page

**Comment by DavideD** [comment]
Added a commit.
I've merged the "How to get help" and "How to contribute sections".
I didn't add the "How to get help" section to the FAQ because there is already a contact us in the beginning of the page that links to the community page.

**Comment by emmanuelbernard** [comment]
> I didn't add the "How to get help" section to the FAQ because there is already a contact us in the beginning of the page that links to the community page.

@DavideD I think it would be better to add it anyways. One indirection and you lose 40% of the people

**Comment by DavideD** [comment]
Added a couple of commits:
- Add How do I get help to the FAQ
- Fix links to the mailing list

**Comment by emmanuelbernard** [comment]
My apologies @DavideD for taking so long. I tried to apply it but it conflicts with production and and can't quite make the diff out.

Also, the text from the contribute file is a quasi copy from the reference documentation contribute chapter. So we probably should apply similar changes.

**Comment by Sanne** [comment]
What's the status here @DavideD @emmanuelbernard ?

**Comment by DavideD** [comment]
> What's the status here @DavideD @emmanuelbernard ?

I don't remember, I'll have a look and let you know

**Comment by yrodiere** [comment]
@DavideD Should we close this?

**Comment by DavideD** [comment]
I think we can close it.  The information should still be relevant so it might be helpful if somebody else is working on this.

**Comment by gsmet** [comment]
IIRC, there were a lot of useful improvements in this one.

Don't merge it but don't close it for now, we need to include these information either in the CONTRIBUTING.md file or the /contribute/ page and I would really prefer we don't forget about it.

**Pull request closed without merging**